### ISSUE TITLE
Update json version for minimal-mistake-jekyll's dependencies

### ISSUE THREAD

**Issue by markhkim** [mmistakes/minimal-mistakes#756]
Installing the newest version of the `minimal-mistakes-jekyll` gem prompts installation of the `json-1.8.3` gem. This version of `json` does not compile with Ruby 2.4, a known issue: https://github.com/flori/json/issues/303

**Comment by markhkim** [comment]
I can confirm that the gem works fine on Ruby 2.3.3.

**Comment by mmistakes** [comment]
This has more to do with Jekyll than the theme per say.

Related: https://github.com/github/pages-gem/issues/376

**Comment by markhkim** [comment]
Hm, does `gem install minimal-mistakes-jekyll` also install `github-pages`? Jekyll installation works fine on Ruby 2.4.0.

**Comment by mmistakes** [comment]
It doesn't, I just included the link because the issue is related. The theme gem doesn't have `json` as a dependency directly. 

I'm pretty sure it's one of the official Jekyll plugins included as dependencies, which is why I included the GH Pages gem link since that too has similar plugin dependencies.

 I've seen this `json` gem problem with core Jekyll too. Just because `gem install jekyll` worries with Ruby 2.4  I wouldn't completely rule it out. The issue I've seen usually creeps up when used with Bundler as this theme does.

Not sure what the fix would be on my end as this appears to be an issue with a dependency I don't control.
